src/plugins/mpath.c

```diff
@@ -15,18 +15,57 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  *
  * Author: Vratislav Podzimek <vpodzime@redhat.com>
+ *         Gris Ge <fge@redhat.com>
+ */
+
+/*
+ * Notes on Code style:
+ *  * 4 spaces instead of tab as indention.
+ *  * Use this format of function block:
+ *      const char *something (int arg1) {
+ *      }
+ *  * Space before '(' unless it's in macro define.
+ *  * Use this format for if/switch/for block:
+ *      if (foo != boo) {
+ *          do_something ();
+ *      }
  */
 
 #include <glib.h>
-/* provides MAJOR, MINOR macros */
-#include <linux/kdev_t.h>
-#include <libdevmapper.h>
+#include <assert.h>
 #include <unistd.h>
 #include <blockdev/utils.h>
+#include <libdmmp/libdmmp.h>
+#include <string.h>
+#include <stdio.h>
 
 #include "mpath.h"
 #include "check_deps.h"
 
+#define _ERR_MSG_BUFF_SIZE    4096
+
+/*
+ * Just don't print to stdout or stderr, and save last error message
+ */
+static void _log_func (struct dmmp_context *ctx, int priority,
+                       const char *file, int line, const char *func_name,
+                       const char *format, va_list args);
+
+static const char *_dmmp_last_err_msg (struct dmmp_context *ctx);
+
+#define _dmmp_ctx_setup(ctx, err_msg_buff) \
+    do { \
+        ctx = dmmp_context_new (); \
+        if (ctx != NULL) \
+          { \
+            memset(err_msg_buff, 0, _ERR_MSG_BUFF_SIZE); \
+            dmmp_context_userdata_set (ctx, err_msg_buff); \
+            dmmp_context_log_func_set (ctx, _log_func); \
+          } \
+    } while(0)
+
+#define _UNUSED(x) (void)(x)
```

Comment:
Why do all these things start with an underscore?

Reply:
Just don't want to override existing macros (from other header) by accident.
Just a habit of keep internal functions start with an underscore.